UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentFileShareService.cs

```diff
@@ -0,0 +1,125 @@
+                        currentProductSub.ProductName = item.ProductName;
+                        currentProductSub.FileSize = item.FileSize;
+                        currentProductSub.Cancellation = item.Cancellation;
+                        listSubUpdateNumberProduts.Add(currentProductSub);
+                    }
+                }
+            }
+            return listSubUpdateNumberProduts;
+        }
+
+        public async Task<List<FulfillmentDataResponse>> QueryFileShareServiceData(List<Products> products)
+        {
+            if (products != null && products.Any())
+            {
+                var batchProducts = SliceFileShareServiceProducts(products);
+                var listBatchDetails = new List<BatchDetail>();
+                foreach (var item in batchProducts)
+                {
+                    var result = await fileShareService.GetBatchInfoBasedOnProducts(item);
+                    listBatchDetails.AddRange(result.Entries);
+                }
+
+                return SetFaltteningFulfillmentData(new SearchBatchResponse()
+                {
+                    Entries = listBatchDetails
+                }); 
+            }
+            return null;
+        }
+
+        public IEnumerable<List<Products>> SliceFileShareServiceProducts(List<Products> products)
+        {
+            return SplitList((SliceFileShareServiceProductsWithUpdateNumber(products)), fileShareServiceConfig.Value.ProductLimit);
+        }
+
+        public static IEnumerable<List<T>> SplitList<T>(List<T> products, int nSize)
+        {
+            for (int i = 0; i < products.Count; i += nSize)
+            {
+                yield return products.GetRange(i, Math.Min(nSize, products.Count - i));
+            }
+        }
+
+        public async Task<string> UploadFileShareServiceData(string uploadFileName, List<FulfillmentDataResponse> fulfillmentDataResponse, string storageAccountConnectionString, string containerName)
+        {
+            var serializeJsonObject = JsonConvert.SerializeObject(fulfillmentDataResponse);
+
+            var cloudBlockBlob = azureBlobStorageClient.GetCloudBlockBlob(uploadFileName, storageAccountConnectionString, containerName);
+            cloudBlockBlob.Properties.ContentType = CONTENT_TYPE;
+
+            using (var ms = new MemoryStream())
+            {
+                LoadStreamWithJson(ms, serializeJsonObject);
+                await azureBlobStorageClient.UploadFromStreamAsync(cloudBlockBlob, ms);
+            }            
+            return cloudBlockBlob.Uri.AbsoluteUri;
+        }
+
+        private void LoadStreamWithJson(Stream ms, object obj)
+        {
+            StreamWriter writer = new StreamWriter(ms);
+            writer.Write(obj);
+            writer.Flush();
+            ms.Position = 0;
+        }
+
+        private List<FulfillmentDataResponse> SetFaltteningFulfillmentData(SearchBatchResponse searchBatchResponse)
+        {
```

Comment:
Should this method be SetFlatteningFulfillmentData

Reply:
renamed method name